mysql-test/suite/galera/t/MDEV-27713.test

```diff
@@ -0,0 +1,54 @@
+--source include/galera_cluster.inc
+--source include/have_innodb.inc
+
+CREATE TABLE t1 (
+  f1 INT,
+  f2 VARCHAR(255) PRIMARY KEY
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+INSERT INTO t1 VALUES(1, 'abc');
+
+--connection node_1
+SET AUTOCOMMIT=OFF;
+START TRANSACTION;
+INSERT INTO t1 VALUES (2,'def');
+
+--connection node_2
+
+SET GLOBAL event_scheduler=ON;
+
+DELIMITER |;
+CREATE PROCEDURE update_table()
+BEGIN
+  SET AUTOCOMMIT=OFF;
+  DO GET_LOCK('local_lock', 0);
+  SET DEBUG_SYNC = 'innodb_row_update_for_mysql_begin SIGNAL blocked WAIT_FOR continue';
+  UPDATE t1 SET f2 = 'jkl' WHERE f1 != 2;
+  DO RELEASE_LOCK('local_lock');
+END|
+DELIMITER ;|
+
+CREATE DEFINER=current_user 
+  EVENT event
+  ON SCHEDULE AT CURRENT_TIMESTAMP
+  ON COMPLETION PRESERVE
+  ENABLE
+  DO CALL update_table();
+
+--connect node_2a, 127.0.0.1, root, , test, $NODE_MYPORT_2
+SET DEBUG_SYNC = 'now WAIT_FOR blocked';
+
+--connection node_1
+COMMIT;
+
+--connection node_2a
+SET DEBUG_SYNC = 'now SIGNAL continue';
+
```

Comment:
I see a potential hazard of non-determinism here. Note that we don't know when the transaction just committed in node_1 will eventually start applying in node_2. If applying is delayed, then this event execution can complete successfully, without any interference with the applier. The UPDATE itself finds no rows in the table, and there will be nothing to commit, nor replicate from the event execution.

Reply:
Updated MTR to make it deterministic